### DISCUSSION TITLE
Find places in government databases

### DISCUSSION THREAD

**Discussion by brawer** [alltheplaces/alltheplaces#7624] [Ideas]
Should AllThePlaces try to mine businesses from government databases, even if those databases aren’t primarily geographic?

In the context of detecting places that have gone out of business, [@1ec5](https://github.com/1ec5) wrote on the [OSM-US slack channel]( https://osmus.slack.com/archives/CDJ4LKA2Y/p1676828639364919?thread_ts=1676579375.006379&cid=CDJ4LKA2Y):

> If you’re focusing on a particular metro area and a particular kind of establishment like restaurants, consider making government inspection or registration data more readily accessible to mappers who are reviewing POIs, even if you don’t go as far as to import that data. For example, any establishment serving food or alcohol, accepting WiC [food stamps] or Medicare, or providing child care services would be in a public database somewhere. If a mapper sees a new entry at the same address as the old OSM POI, there’s probably been turnover.

For example, the San Francisco Department of Public Health publishes [health inspection records for 53K businesses](https://data.sfgov.org/widgets/pyih-qa8i) that are legally required to comply with food safety regulations. AllThePlaces could fetch such government lists once per week and convert them to a standard format (GeoJSON) with a standard schema (OpenStreetMap tags). (Comparing a sequence of weekly dumps to find closed businesses would be an interesting follow-up problem, but probably out of scope for the AllThePlaces project). Unfortunately, this kind of government data tends to be rather noisy, sparsely geocoded if at all, and it typically lacks properties that are important for OpenStreetMaps. For example, the San Francisco public health dataset doesn’t distinguish bars from grocery stores.

Would AllThePlaces consider such government data in scope for the project? Personally, I won’t have time to do much myself, but when I saw @1ec5’s posts on Slack, I thought why not start a discussion.

**Comment by iandees** [comment]
My initial reaction is that that government records would fall under my desire to not include data aggregators. Health inspection records, building permits, etc. are not primary sources for place information. They're definitely useful for checking if a place exists in OSM, but I'm inclined to say they don't fit in ATP.

**Comment by Cj-Malone** [comment]
So what was described is basically Food Hygiene Ratings System (`fhrs:id`) from Food Standards Agency in GB. We then have a [tool](https://gregrs.dev.openstreetmap.org/fhodot/?lat=53.38209&lon=-1.46899&zoom=17&layer=suggest) that compares that data to OSM, suggests match where we add the ref and address, warns when a ref no longer exists or OSM has a POI that should be in the gov data but isn't.

If we could unify the data from governments, and unify a tool like that to help improve OSM globally. It would massively improve OSM. And "we" should help achieve that, whether it's in ATP (With dataset_attributes, thank you) or a "sister project", I don't massively care. But standardising datasets in OSM tagging will help raise awareness for OSM, and help data consumers use more Open Data. And spread the ODbL. (That's my goal, I don't know about others.)

**Comment by Cj-Malone** [comment]
I think we actually already do this, we just haven't realised it.

We already have city level open data from @brawer, we have NHS data from @mjoe999 and we have bus stations from me. Here would be a good place to parse FHRS which is already parsed by several people in the GB community (@rjw62 probably knows a bunch more GB datasets). I'm sure [AU Schools](https://data.cese.nsw.gov.au/data/dataset/nsw-public-schools-master-dataset) would help @davidhicks and any code for the US GNIS thing could belong here.

I think the only thing to discus are the details. They probably don't need to be ran weekly? Split them into a different directory?

If we exclude them from the weekly, the only thing that changes is we hopefully have a few more OSM people around.

**Comment by brawer** [comment]
Curious, in what respect would splitting off government data from chains make a difference?

Regarding the frequency of running spiders, the upstream change rate appears to be similar to chain stores. For example, in the case of the City of Zürich, the upstream data can theoretically change constantly (afaik their open data portal is basically a read-only mirror of their live municipal GIS database, which gets updated all the time). In reality, there isn't a new playground or recycling bin every day, but looking at the weekly stats of past ATP runs, there have been changes in many but not all weeks — just as with chains. Of course, that'll look different for governments that populate their data portals with a manual process. But is there a disadvantage with keeping a weekly schedule?

Regarding legal concerns, isn't that already addressed by explicitly tagging the license of each dataset? Like stores, different cities use different licenses, and many are not stating explicitly any data license at all.

Are we concerned about data volume? So far, these datasets are not hugely different from large chains. Even if we were to import buildings and trees (so far we didn't bother), it would only be a couple 100K features per city.

Said this, personally I have no strong opinion either way. I just find it immensely useful that ATP offers all the data in the same format and schema, and keeps everything reasonably fresh.

**Comment by iandees** [comment]
I think rather than splitting off some spiders into a separate directory we should think about a system that runs periodically and only updates spiders that have a history of changing. Then the weekly grabs the latest data. The government sources could be marked as "no updates" or something.

**Comment by davidhicks** [comment]
@Cj-Malone On NSW schools data, I have been thinking of what limits this project to brands and not operators as name-suggestion-index also captures. I don't think any other project has been creating spiders for scraping electrical substations, fire stations, schools, etc, so there appears to be a demand and a gap to fill.

**Reply by CloCkWeRX** [reply]
https://opencouncildata.org/ made some progress for common datasets but never quite made it out of Victoria. https://opencouncildata.github.io/Platform/

**Comment by matkoniecz** [comment]
Note that feeding government business data was tried by some map data providers. At least in Poland it results in flood of nonexisting offices.

Many people gave single-person company without any office but registered at some address. For example I have seen "logistics company office" that is actually where some trucker lives, there is no office, company is existing only to keep taxes saner.

Government data is especially prone to listing nonexisting stuff that legally exists and not listing things that legally do not exist or have no presence. Compare "registered address of a company" and "company is operating POI at location XYZ".

Or bizarre legal names, drastically different from signposted names.

**Reply by 1ec5** [reply]
Yes, government data can be problematic if you don’t vet the source well. Several years ago, Apple Maps infamously [ingested business listings](https://www.theterminal.info/i/39946708/the-sunken-land-rises-again) from an Australian government agency and it [littered the map of Sydney with retirement trusts](https://web.archive.org/web/20200622043959/https://twitter.com/mxswd/status/1020486517989392384) on otherwise unremarkable residences. Other sources consist of data contributed by residents or shopkeepers, which you need to clean up heavily. I’ve seen that firsthand working on the neverending [Silicon Valley POI import](https://wiki.openstreetmap.org/wiki/Santa_Clara_County,_California/Social_distancing_protocol_import).

On the other hand, if you use the data correctly, it can be a boon for the map. Where I live, restaurant inspection data would be quite good if we can reliably distinguish between fast-food and sit-down establishments. There’s even a database of [any business that deals with hazardous waste](https://hwts.dtsc.ca.gov/), such as car repair shops. I’m eyeing that one for an import into OpenHistoricalMap because the recordkeeping is so comprehensive and the coordinates are reasonably accurate.

**Comment by CloCkWeRX** [comment]
Worth moving to Discussions?
Useful topic, but also hard to decide when it is "done"
